docs/source/coding_style.md

```diff
@@ -0,0 +1,243 @@
+# INC Coding Conventions
+
+ (Mostly for Version 3 and later)
+
+## Background
+
+To improve the quality and maintainability of INC code, we summarized some common coding standards and conventions.
+
+There are many style guides, and they may conflict with each other. To avoid overly arguing formatting, we make decisions based on the following priorities:
+
+- [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings), [PEP 8](https://peps.python.org/pep-0008/)
+- Framework Style
+- INC Internal Style
+- Sub-module specific Style
+
+## Rules
+
+> Note: The sub-tile naming is following [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings) and [PEP 8](https://peps.python.org/pep-0008/). See the relevant section for more details.
+
+
+### Imports
+
+- Recommend
+
+```python
+import os
+import sys
+
+from x import py
+from x import y as z
+from copy import deepcopy
+from subprocess import Popen, PIPE
+```
+
+- Not recommend
+
+```python
+from sub_module import *  # May lead to namespace pollution
+import os, sys  # Import on separate lines
+import copy  # May import local copy.py
+```
+
+### Strings
+
+- Recommend
+
+```python
+long_string = """This is fine if your use case can accept
+    extraneous leading spaces."""
+
+long_string = "And this is fine if you cannot accept\n" "extraneous leading spaces."
+```
+
+- Not recommend
+
+```python
+logger.info("This is fine if your use case can accept")
+logger.info("extraneous leading spaces.")
+```
+
+### Logger
+
+- Recommend
+
+```python
+from neural_compressor.common import logger
+
+logger.info("Current TensorFlow Version is: %s", tf.__version__)  # Use a pattern-string (with %-placeholders)
+
+logger.info("Current $PAGER is: %s", os.getenv("PAGER", default=""))  # Better readability
+
+# Handle long string
+logger.warning(
+    "All tuning options for the current strategy have been tried. \n"
+    "If the quantized model does not seem to work well, it might be worth considering other strategies."
+)
+
+logger.warning(
+    "This is a long string, this is a long string,"
+    "override the user config's smooth quant alpha into the best alpha(%.4f) found in pre-strategy.",
+    0.65421,
+)
+```
+
+- Not recommend
+
+```python
+logger.info(f"Current TensorFlow Version is: {tf.__version__}")  # Not use f-string
```

Comment:
little confusing,  it is better to change to "use f-string" ,  it will be consistent that you use comment to show the format problem similar in other case like "One sentence in two lines"

Reply:
Get it, thanks. updated it in [82eea64](https://github.com/intel/neural-compressor/pull/1584/commits/82eea643997982313676ea1b3370d38caa00fc72).